node/src/chain/fsm.rs

```diff
@@ -28,7 +26,8 @@ use tracing::{debug, error, info, warn};
 
 const MAX_BLOCKS_TO_REQUEST: i16 = 50;
 const EXPIRY_TIMEOUT_MILLIS: i16 = 5000;
-
+const DEFAULT_HOPS_LIMIT: u16 = 100;
+const DEFAULT_CERT_CACHE_EXPIRY: Duration = Duration::from_secs(10);
```

Comment:
> This means after I add the certificate to the cache it might take 10 seconds to receive the next block.

There is a misunderstanding here.

The received Quorum is for Candidate/Block the network has already reached consensus on. That said, 10s is the time we expect the request to `travel` and a valid response(s) to be received.

Reply:
Yeah, I get that. My point is that a response to that request might still be useful after 10 seconds

Reply:
>Yeah, I get that. My point is that a response to that request might still be useful after 10 seconds

For my benefit, could you then clarify in what other scenarios in the context of this PR/issue we will need to keep it? 

If this will speed up the PR review, lemme know what number is suitable here?

Reply:
Let's say we receive Quorum for block 500; we put the certificate in the cache and ask the candidate around.
Then we receive the next Candidate 501. 
After 10 seconds, we receive the Candidate for block 500, and accept it. After 11 seconds we receive the Quorum for 501. We can now accept 501 with no further delay.

If instead the certificate expired after 10 seconds, when receiving the Quorum for 501 we would consider ourselves as out of sync and ask our peer for the missing block 500.

This is my understanding at least. Forgive me if I'm getting this wrong.

What I'm trying to say is that I don't see any "rush" in deleting certificates from cache, while I can see they can be useful as long as we might receive the corresponding Candidate. If we're not interested in this Candidate anymore (e.g., because we detect it's from a lower-priority fork), than we should delete the certificate when this is detected.
Again, I'm trying to decouple consensus logic from time constraints and rely more on consensus-related information.

In my opinion, this expiration time should be just an emergency cleanup to do in case we don't delete the certificates by other means (receiving the corresponding candidate, detecting a fork, ...).
I would set it to something high, like 1 minute, or even more.

Reply:
>After 10 seconds, we receive the Candidate for block 500, and accept it

Usually, the candidate is expected to be received for around ~200-400ms. 10sec delay in response for a flood request is insane.

> After 11 seconds we receive the Quorum for 501. We can now accept 501 with no further delay

That's a bit inaccurate. If the node really waits 10sec for  block 500 to come, it's more probable to receive block 501 before 500. If block 501 comes before 500, it will be discarded.   

> I would set it to something high, like 1 minute, or even more.
The higher the settings is the more vulnerable is the cache to spam attack is (an adversary spamming a node with fake quorums)

Anyway, to make things flow, I'll change it to 1min after all.

Reply:
For the record, our protocol assumes, at best, a partially synchronous model. So no delay or loss is insane and should be considered possible. 
If we were in a synchronous model we would probably use a different consensus protocol.

As for spamming fake quorums I really don't think that should be an issue, assuming we do a minimum safety check on them.

Reply:
The "consensus protocol" is not relevant to the delay here. That's simply a request-response message flow in a distributed manner that depends mainly on the network latency (assuming nodes are on multi-core machines thus processing requests unblocked.)